### PR TITLE
Enhancement/3586 improve workflow

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #3586

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/analytics/components/settings/SettingsEdit.test.js [hunk 1]
```diff
@@ -20,18 +20,26 @@
...
 describe( 'SettingsEdit', () => {
 	let registry;
 
 	beforeEach( () => {
 		registry = createTestRegistry();
 		provideSiteInfo( registry );
+		provideModules( registry, [
+			{
+				slug: 'analytics',
+				activate: true,
```
Comment (aaemnnosttv) [inline assets/js/modules/analytics/components/settings/SettingsEdit.test.js:39]: This should be `active` instead.
```suggestion
				active: true,
```

File: assets/js/modules/analytics/datastore/accounts.test.js [hunk 2]
```diff
@@ -485,6 +486,18 @@ describe( 'modules/analytics accounts', () => {
 					} );
 
 					provideSiteInfo( registry );
+					provideModules( registry, [
+						{
+							slug: 'analytics',
+							activate: true,
```
Comment (aaemnnosttv) [inline assets/js/modules/analytics/datastore/accounts.test.js:492]:
```suggestion
							active: true,
```

File: assets/js/modules/analytics/datastore/accounts.test.js [hunk 3]
```diff
@@ -485,6 +486,18 @@ describe( 'modules/analytics accounts', () => {
...
+							connected: false,
+						},
+						{
+							slug: 'analytics-4',
+							activate: true,
```
Comment (aaemnnosttv) [inline assets/js/modules/analytics/datastore/accounts.test.js:497]:
```suggestion
							active: true,
```

**Review by aaemnnosttv — COMMENTED** [review]
Looks good to me overall, just a few typos in a few of the tests which I'll apply now.

File: assets/js/modules/analytics/components/settings/SettingsForm.js [hunk 4]
```diff
@@ -86,18 +81,14 @@ export default function SettingsForm() {
...
+				<GA4PropertyNotice notice={ __( 'You need to connect the Google Analytics 4 property that’s associated with this Universal Analytics property.', 'google-site-kit' ) }>
 					<div className="googlesitekit-setup-module__inputs">
 						<GA4PropertySelect />
 					</div>
 				</GA4PropertyNotice>
 			) }
 
-			{ ( isGA4Enabled && SETUP_FLOW_MODE_UA === setupFlowMode ) && (
-				<GA4PropertyNotice notice={ __( 'A Google Analytics 4 property will also be created.', 'google-site-kit' ) } />
-			) }
```
Comment (aaemnnosttv) [inline assets/js/modules/analytics/components/settings/SettingsForm.js:99]: Should this be removed?

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @eugene-manuilov – at least one thing I noticed which still needs addressing.

File: assets/js/modules/analytics/components/settings/SettingsForm.js [hunk 5]
```diff
@@ -86,18 +81,14 @@ export default function SettingsForm() {
 				</div>
 			) }
 
-			{ ( isGA4Enabled && ( SETUP_FLOW_MODE_GA4_TRANSITIONAL === setupFlowMode || SETUP_FLOW_MODE_GA4 === setupFlowMode ) ) && (
-				<GA4PropertyNotice notice={ __( 'You’ll need to connect the Google Analytics 4 property that’s associated with this Universal Analytics property.', 'google-site-kit' ) }>
+			{ ( isGA4Enabled && !! ga4PropertyID ) && (
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/modules/analytics/components/settings/SettingsForm.js]: Yes, it's a good idea. Updated to use the `isModuleConnected` selector.

File: assets/js/modules/analytics/components/settings/SettingsForm.js (hunk 4 above)
Comment (eugene-manuilov) [reply assets/js/modules/analytics/components/settings/SettingsForm.js]: I think you are right, this shouldn't be removed. Reverted it.

**Review by aaemnnosttv — COMMENTED** [review]
@eugene-manuilov One more thing that appears left to be addressed from the ACs:

> Connected to the above, the UI of the Analytics `SettingsEdit` view should be updated:
> * If a saved GA4 property is set (on page load), instead of displaying the same notice about selecting an associated GA4 property that is present in the setup flow, _just_ the GA4 property dropdown itself should be displayed, without any further information (since at this point the user already set up GA4 so should have context).
>     * The label of the GA4 property dropdown should _only in this case_ be adjusted to "Google Analytics 4 Property" instead of just "Property".

image

Note: I changed a few selections here to not show other IDs but the display is the same.

**Comment by aaemnnosttv** [comment]
@googlebot I consent.

**Comment by eugene-manuilov** [comment]
Thanks, @aaemnnosttv. PR is updated.

File: assets/js/modules/analytics/components/settings/SettingsForm.js [hunk 6]
```diff
@@ -86,16 +81,14 @@ export default function SettingsForm() {
...
 			) }
 
-			{ ( isGA4Enabled && ( SETUP_FLOW_MODE_GA4_TRANSITIONAL === setupFlowMode || SETUP_FLOW_MODE_GA4 === setupFlowMode ) ) && (
-				<GA4PropertyNotice notice={ __( 'You need to connect the Google Analytics 4 property that’s associated with this Universal Analytics property.', 'google-site-kit' ) }>
-					<div className="googlesitekit-setup-module__inputs">
-						<GA4PropertySelect />
-					</div>
-				</GA4PropertyNotice>
+			{ ( isGA4Enabled && isGA4Connected ) && (
+				<div className="googlesitekit-setup-module__inputs">
```
Comment (aaemnnosttv) [inline assets/js/modules/analytics/components/settings/SettingsForm.js:85]: I think maybe we should add a class here to place the select below in the same position as the input for the new view name. It's a little odd that this one sits lower than that one.

**Review by aaemnnosttv — COMMENTED** [review]
Looks good to me, just one last thing I noticed about the spacing between the top inputs and the GA4 property below.

File: assets/js/modules/analytics/components/settings/SettingsForm.js (hunk 6 above)
Comment (eugene-manuilov) [reply assets/js/modules/analytics/components/settings/SettingsForm.js]: Updated.

**Review by aaemnnosttv — APPROVED** [review]
Great, thanks!

VRT is failing, but I'll open a PR for that separately since it's not related to these changes.

**Pull request merged**